src/main/java/com/google/firebase/projectmanagement/AndroidApp.java

```diff
@@ -0,0 +1,191 @@
+/* Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.projectmanagement;
+
+import com.google.api.core.ApiFuture;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import java.util.List;
+
+/**
+ * An instance of this class is a reference to an Android App within a Firebase Project; it can be
+ * used to query detailed information about the App, modify the display name of the App, or download
+ * the configuration file for the App.
+ */
+public class AndroidApp {
+
+  private final AndroidAppService androidAppService;
+  private final String appId;
+
+  private AndroidApp(String appId, AndroidAppService androidAppService) {
```

Comment:
An example of a thing that can be done in a factory method that you cannot do in a constructor is use a different method name for two different factory methods. The reason this is needed is we plan to have another constructor in our upcoming proposal that is also based on a String (the fully qualified resource name of the app, rather than the App ID). This is impossible with a constructor since you cannot overload the constructor with exactly the same signature.